File: es6/hero.mjs
export class Hero {

  constructor(name, mainhand, offhand) {
    this.name = name;
    this.mainhand = mainhand;
    this.offhand = offhand;
  }

  toString() {
    return `${this.name} (${this.mainhand}, ${this.offhand})`;
  }

  lmb() {
    return (`${this.name} uses ${this.mainhand}!`);
  }

  rmb() {
    return (`${this.name} uses ${this.offhand}!`);
  }
}

File: rust/src/hero.rs
use crate::item::Item;

#[derive(Debug)]
pub struct Hero {
    pub name: String,
    pub mainhand: Item,
    pub offhand: Item,
}

impl Hero {
    pub fn lmb(&self) -> String {
        return format!("{} uses {}!", &self.name, &self.mainhand.name);
    }
    pub fn rmb(&self) -> String {
        return format!("{} uses {}!", &self.name, &self.offhand.name);
    }
}
